currency-rates/internal/repo/subsctiption.go

```diff
@@ -8,19 +8,25 @@ import (
 
 type DBEmail struct {
     Email string
+	IsActive bool
 }
 
 func (repo *Repository) SetEmail(ctx context.Context, email string) error {
 	e := &DBEmail{
 		Email: email,
+		IsActive: true,
 	}
 
 	return repo.DB.Table("emails").Create(e).Error
 }
 
 func (repo *Repository) GetEmails(ctx context.Context) ([]services.Email, error) {
 	result := make([]services.Email, 0)
-	err := repo.DB.Table("emails").Find(&result).Error
+	err := repo.DB.Table("emails").Where("is_active = ?", true).Find(&result).Error
 	
 	return result, err
 }
+
+func (repo *Repository) DeactivateEmail(ctx context.Context, email string) error {
+	return repo.DB.Model(services.Email{}).Where("email = ?", email).Update("is_active", false).Error
+}
```

Comment:
Seems more like businnes logic, not mechanism of saving. Look of this like deactivating email and save it state. So maybe you can just deactivate it in domain and just call method Save to store actual state of email?

Reply:
Done